electronics_lib/DcDcConverters.py

```diff
@@ -1,5 +1,85 @@
 from electronics_abstract_parts import *
 
+class Tps61023_Device(DiscreteChip, FootprintBlock):
+  @init_in_parent
+  def __init__(self, current_draw: RangeLike = RangeExpr()):
+    super().__init__()
+    self.vin = self.Port(VoltageSink(
+      voltage_limits=(0.5, 5.5)*Volt,
+      current_draw=current_draw
+    ))
+    self.gnd = self.Port(Ground())
+    self.sw = self.Port(VoltageSource())
+    self.fb = self.Port(AnalogSink(impedance=(8000, float('inf')) * kOhm))
+    self.vout = self.Port(VoltageSource())
+
+  def contents(self) -> None:
+    super().contents()
+    self.footprint(
+      'U', 'Package_TO_SOT_SMD:SOT-23-6',
+      {
+        '1': self.fb,
+        '2': self.vin, # en
+        '3': self.vin,
+        '4': self.gnd,
+        '5': self.sw,
+        '6': self.vout,
+      },
+      mfr='Texas Instruments', part='TPS61023',
+      datasheet='https://www.ti.com/lit/ds/symlink/tps61023.pdf'
+    )
+
+class Tps61023(FootprintBlock):
+  @init_in_parent
+  def __init__(self) -> None:
+    super().__init__()
+    self.gnd = self.Port(Ground())
+    self.vin = self.Port(VoltageSink(
+      voltage_limits=(0.5, 5.5)*Volt,
+      current_draw=(0, 30) * uAmp
+    ))
+    self.vout = self.Port(VoltageSource())
+
+  def contents(self) -> None:
+    super().contents()
+    self.dcdc_converter = self.Block(Tps61023_Device())
+
+    self.vin_cap = self.Block(Capacitor())
+    self.vout_cap = self.Block(Capacitor())
+    self.ind = self.Block(Inductor())
+    self.vout_r1 = self.Block(Resistor())
+    self.vout_r2 = self.Block(Resistor())
+
+    self.footprint(
```

Comment:
Why does this have a footprint? Footprints should be physical components only, I don't think there's a 3-pin component that's part of this circuit?